x/ccv/provider/keeper/relay_test.go

```diff
@@ -469,8 +473,92 @@ func TestHandleSlashPacket(t *testing.T) {
+		providerKeeper.SetInitChainHeight(ctx, chainId, 5)
+		providerKeeper.SetValsetUpdateBlockHeight(ctx, validVscID, 99)
+		// Setup consumer address to provider address mapping.
+		providerKeeper.SetValidatorByConsumerAddr(ctx, chainId, sdk.ConsAddress(tc.packetData.Validator.Address), providerConsAddr)
+
+		// Execute method and assert expected mock calls.
+		providerKeeper.HandleSlashPacket(ctx, chainId, tc.packetData)
+		ctrl.Finish()
+	}
+}
+
+// TestHandleSlashPacket tests the handling of slash packets.
+func TestSlashAckAppendAfterHandleSlashPacket(t *testing.T) {
+
+	chainId := "consumer-id"
+	validVscID := uint64(234)
+	consumerPubKey := ed25519.GenPrivKey().PubKey()
+	consumerConsAddr := sdk.ConsAddress(consumerPubKey.Address())
+
+	providerPubKey := ed25519.GenPrivKey().PubKey()
+	providerConsAddr := sdk.ConsAddress(providerPubKey.Address())
+
+	testCases := []struct {
+		name       string
+		packetData ccv.SlashPacketData
+		// The mocks that we expect to be called for the specified packet data.
+		expectedCalls        func(sdk.Context, testkeeper.MockedKeepers, ccv.SlashPacketData) []*gomock.Call
+		expectedSlashAcksLen int
+	}{
+		{
+			"full downtime packet handling, uses init chain height and non-jailed validator",
+			*ccv.NewSlashPacketData(abci.Validator{Address: consumerConsAddr.Bytes()}, 0, stakingtypes.Downtime), // ValsetUpdateId = 0 uses init chain height.
+			func(ctx sdk.Context, mocks testkeeper.MockedKeepers,
+				expectedPacketData ccv.SlashPacketData,
+			) []*gomock.Call {
+				return testkeeper.GetMocksForHandleSlashPacket(
+					ctx, mocks, expectedPacketData, providerConsAddr, stakingtypes.Validator{Jailed: false})
 			},
+			1,
+		},
+		{
+			"full downtime packet handling, uses valid vscID and jailed validator",
+			*ccv.NewSlashPacketData(abci.Validator{Address: consumerConsAddr.Bytes()}, validVscID, stakingtypes.Downtime),
+			func(ctx sdk.Context, mocks testkeeper.MockedKeepers,
+				expectedPacketData ccv.SlashPacketData,
+			) []*gomock.Call {
+				return testkeeper.GetMocksForHandleSlashPacket(
+					ctx, mocks, expectedPacketData, providerConsAddr, stakingtypes.Validator{Jailed: true})
+			},
+			1,
+		},
+		{
+			"full double sign packet handling, uses init chain height and jailed validator",
+			*ccv.NewSlashPacketData(abci.Validator{Address: consumerConsAddr.Bytes()}, 0, stakingtypes.DoubleSign),
+			func(ctx sdk.Context, mocks testkeeper.MockedKeepers,
+				expectedPacketData ccv.SlashPacketData,
+			) []*gomock.Call {
+				return testkeeper.GetMocksForHandleSlashPacket(
+					ctx, mocks, expectedPacketData, providerConsAddr, stakingtypes.Validator{Jailed: true})
+			},
+			0, // no slash acks for double sign - only tombstone
+		},
+		{
+			"full double sign packet handling, uses valid vsc id and non-jailed validator",
+			*ccv.NewSlashPacketData(abci.Validator{Address: consumerConsAddr.Bytes()}, validVscID, stakingtypes.DoubleSign),
```

Comment:
We don't need to test this method for double-sign infractions, it'll only be called with downtime slash packets. might be worth changing the name of the method in the future